File: algorithm/src/BOJ2609.java
/**
 * @author nakhoon
 * @date 2022/05/29
 * @see https://www.acmicpc.net/problem/2609
 * @mem 11,496kb
 * @time 76ms
 * @caution
 * [고려사항]
 * 두 수의 곱 / 최대공약수 = 최소공배수의 공식을 이용해서 최대공약수와 최소공배수를 출력하였다.
 * [입력사항]
 * [출력사항]
 */
import java.util.*;
import java.io.*;
//백준 <수학> '최대공약수와 최소공배수'
public class BOJ2609 {
  public static void main(String[] args) throws IOException{
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = new StringTokenizer(br.readLine());
    int N = Integer.parseInt(st.nextToken());
    int M = Integer.parseInt(st.nextToken());

    System.out.println(GCD(N,M));
    System.out.println(LCM(N,M));
  }

  static int GCD(int a, int b){
    while(b > 0){
      int temp = a;
      a = b;
      b = temp%b;
    }

    return a;
  }

  static int LCM(int a, int b){
    return (a*b)/GCD(a,b);
  }
}

File: algorithm/src/BOJ2609.kt
/**
 * @author nakhoon
 * @date 2024/01/18
 * @see https://www.acmicpc.net/problem/2609
 * @mem 12,228kb
 * @time 88ms
 * @caution
 * [고려사항]
 * [입력사항]
 * [출력사항]
 */
import java.io.*
import java.util.*

//백준 <수학> '최대공약수와 최소공배수'
fun main(args: Array<String>) = with(BufferedReader(InputStreamReader(System.`in`))) {
    val st = StringTokenizer(readLine())
    val A = Integer.parseInt(st.nextToken())
    val B = Integer.parseInt(st.nextToken())
    val gcd = getGcd(A, B)

    println(gcd)
    println(A * B / gcd)
}

fun getGcd(a:Int, b: Int):Int{
    var num1 = a
    var num2 = b
    while (num2 > 0) {
        val temp = num1
        num1 = num2
        num2 = temp % num2
    }

    return num1
}